tests/src/test/scala/com/nvidia/spark/rapids/QueryFuzzer.scala

```diff
@@ -0,0 +1,240 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.nvidia.spark.rapids
+
+import java.util.concurrent.atomic.AtomicLong
+
+import scala.util.Random
+
+import org.apache.spark.SparkConf
+import org.apache.spark.sql.{DataFrame, SparkSession}
+import org.apache.spark.sql.functions.col
+import org.apache.spark.sql.internal.SQLConf
+import org.apache.spark.sql.types.{DataTypes, DecimalType}
+
+class QueryFuzzer(val seed: Long) {
+
+  val rand = new Random(seed)
+
+  val idGen = new AtomicLong(0)
+
+  def generateConfig(): SparkConf = {
+    new SparkConf()
+      .set(SQLConf.ADAPTIVE_EXECUTION_ENABLED.key, boolString())
+      .set(SQLConf.ANSI_ENABLED.key, boolString())
+      .set(SQLConf.LEGACY_TIME_PARSER_POLICY.key, oneOf("LEGACY", "CORRECTED", "EXCEPTION"))
+      .set("spark.plugins", "com.nvidia.spark.SQLPlugin")
+      .set(RapidsConf.DECIMAL_TYPE_ENABLED.key, boolString())
+      .set(RapidsConf.INCOMPATIBLE_DATE_FORMATS.key, boolString())
+      .set(RapidsConf.INCOMPATIBLE_OPS.key, boolString())
+      .set(RapidsConf.ENABLE_CAST_STRING_TO_FLOAT.key, boolString())
+      .set(RapidsConf.ENABLE_CAST_STRING_TO_DECIMAL.key, boolString())
+      .set(RapidsConf.ENABLE_CAST_STRING_TO_DECIMAL.key, boolString())
+      .set(RapidsConf.ENABLE_CAST_STRING_TO_INTEGER.key, boolString())
+      .set(RapidsConf.ENABLE_CAST_STRING_TO_TIMESTAMP.key, boolString())
+      .set(RapidsConf.ENABLE_CAST_FLOAT_TO_STRING.key, boolString())
+      .set(RapidsConf.ENABLE_CAST_FLOAT_TO_DECIMAL.key, boolString())
+      .set(RapidsConf.ENABLE_CAST_FLOAT_TO_INTEGRAL_TYPES.key, boolString())
+      .set(RapidsConf.ENABLE_CAST_DECIMAL_TO_STRING.key, boolString())
+      .set(RapidsConf.ENABLE_FLOAT_AGG.key, boolString())
+      .set(RapidsConf.ENABLE_HASH_OPTIMIZE_SORT.key, boolString())
+  }
+
+  private def boolString(): String = {
+    rand.nextBoolean().toString
+  }
+
+  private def oneOf(value: String*): String = {
+    value(rand.nextInt(value.length))
+  }
+
+  def randomOperator(spark: SparkSession, depth: Int, maxDepth: Int): DataFrame = {
+
+    // if we have reached maximum depth then create a leaf node
+    if (depth == maxDepth) {
+      return generateDataSource(spark)
+
+      //TODO Add support for reading from CSV and Parquet data sources
```

Comment:
Not sure this comment needs to be here.  It's an implementation detail and arguably should be something `generateDataSource` should do.  Also farming from a single, static file is probably not very 'fuzzy', may be better to emit fuzzed values to a temporary Parquet or CSV file and feed that as input.